spec/models/project_duplicator_spec.rb

```diff
@@ -92,4 +91,64 @@
       end
     end
   end
+
+  context 'with scheduled children' do
+    # Creates a timeline and returns nodes stored in a hash.
+    let!(:nodes) { ProjectGroupFactoryHelper.create_full_timeline }
+    let(:loan) { nodes[:root].project }
+
+    # Break each of the nodes out into a let so that we can examine them individually.
+    ProjectGroupFactoryHelper::NODE_NAMES.each do |name|
+      let(name) { nodes[name] }
+    end
+
+    before do
+      # See ProjectGroupFactoryHelper for the layout of nodes in the timeline.
+      # We introduce some schedule dependencies in such a way that one of the dependencies
+      # actually goes downward instead of upward.
+      # g3_s3 is sorted above s1 because g3 has an early step. But we're going to make g3_s3 depend
+      # on s1, which won't change the sort order, but is a downward depenendency.
+      g3_s3.update_attributes!(schedule_parent_id: s1.id)
+      g5_s1.update_attributes!(schedule_parent_id: g3_s3.id)
```

Comment:
Why use the more verbose version of `update`?

Reply:
No reason. Old habits die hard? I'll try to remember next time.